cmake/macros/macro_deal_ii_insource_setup_target.cmake

```diff
@@ -81,7 +81,7 @@ MACRO(DEAL_II_INSOURCE_SETUP_TARGET _target _build)
         )
 
       SET_TARGET_PROPERTIES(${_target} PROPERTIES
-        CUDA_SEPARABLE_COMPILATION TRUE
+        CUDA_SEPARABLE_COMPILATION FALSE
```

Comment:
Since `FALSE` is the default, does it make sense to just remove these lines completely?

Reply:
out of curiosity, is this flag related to 
```
--generate-code arch=compute_xy,code=sm_xy  --generate-code arch=compute_xy,code=compute_xy
```
I had some compiler error while trying to compile dealii with those flags, which were recommended for heterogeneous HPC GPU setup (not that I use one, i was just testing with a single architecture)?

Reply:
@masterleinad I left the code here so it's easier to turn separable compilation on in the future. This way we already know where we need the function. I can remove it if you prefer.

@davydden Maybe :-) This PR fix a bug during linking. So if the error was during the compilation it won't fix it but the flags that you show where not passed to the linking phase. So it could be the problem you saw.

Reply:
Thanks @Rombur , i was not precise enough, i just recall I had some issue building, but I don't remember whether it's at linking or compiling stage. Either way i will try again once this is merged...

Reply:
@davydden As far as I know this is unrelated.

Reply:
@davydden Let's merge this so you can try. :wink:

@Rombur I am fine.